### ISSUE TITLE
[4.1][Bug] Relationship type column not return translatable title

### ISSUE THREAD

**Issue by t-prod** [Laravel-Backpack/CRUD#2647]
When I try to get ProductCategory title is a translatable field. Value is returned without getting the good locale

```
$this->crud->addColumn([
                'name' => 'category',
                'type' => 'relationship',
                'label' => __('admin.product.fields.category'),
                'model' => '\App\Models\ProductCategory',
                'attribute' => 'title',
                'entity' => 'category',
                'visibleInTable' => true
]);
```

Returns

`{"fr":"BOTTES"}`

instead of BOTTES

How can I do ?

**Comment by pxpm** [comment]
Hello @t-prod nice catch. But this was completely lucky by me/you. Heheh

Follow me here, you are trying to add a **Column** but the relationship is made for **Fields**. 
My luck here is that I did not spot it in first place so I endup testing the fields, and realized I didn't pushed the fix accross all relationship fields, had only pushed the fix into fetch_or_create.blade.php. :)

It's now fixed for `fields`. https://github.com/Laravel-Backpack/CRUD/pull/2632/commits/a5fd3ab08162baa8186eb0acbe1e5afdfca3415d

If you think the idea of having the same "relationships" for columns is something worth you can open an issue or a pull request, we are always open to new ideas/code submission.


I am going to close this, as this is really not a bug with relationship, atleast for now until we introduce this in columns :)


Thank you very much. :+1:

**Comment by t-prod** [comment]
Hi pxpm,

Sorry but I think it's not fixed I see your PR and it's OK but not in columns. You can test with a Sluggable Model. If you make a selection 1-n relationship or a relationship as above.
It returns the entire json value not only the locale wished.

Regards

**Comment by pxpm** [comment]
Hey @tabacitu , why did you re-opened this ? 

We don't have a column type relationship. 

You re-opened this because we should do it or .. ?

Best,

**Comment by t-prod** [comment]
Hi pxpm and tabacitu,

Here are more explanations :

`[
    'type'      => "select",
    'label'     => __('admin.product.fields.category'),
    'name'      => 'product_category_id',
    'entity'    => 'category',
    'attribute' => "title",
    'model'     => "App\Models\ProductCategory",
    'tab'   => __('admin.product.tabs.general'),
    'wrapperAttributes' => [
        'class' => 'form-group col-md-6'
    ],
],
`
And in ProductCategory model we have this

`protected $translatable = ['title', 'slug'];`

The result is  the entire json and not only the locale fallback or by default

**Comment by t-prod** [comment]
I think the fix is to do here in this function getRelatedEntriesAttributes in vendor\backpack\crud\src\app\Library\CrudPanel\CrudPanel.php

**Comment by tabacitu** [comment]
Indeed Pedro - I think it's natural to have a ```relationship``` column that shows perfectly what was set using the ```relationship``` field. I think that's what a developer expects. 

Actually, we do have a relationship column in 4.1. [Check it out here](https://backpackforlaravel.com/docs/4.1/crud-columns#relationship).

**Comment by t-prod** [comment]
The relationship column has been tested and same result we have the entire result in JSON wen using spatie sluggable ;)

**Comment by tabacitu** [comment]
Fixed by merging #2670  - thanks a lot for submitting the PR @t-prod , it was PERFECT!
